### ISSUE TITLE
Unhandled error when compilation failed

### ISSUE THREAD

**Issue by bazo** [kotas/gulp-tsc#8]
I use gulp-tsc with watch task.
If i have an error in my ts files and hit save, an unhandled error is thrown and gulp exits.

I believe this error should be caught so that i can fix my code without restarting gulp

**Comment by kotas** [comment]
@bazo Thank you for report!

Yes, it is actually common behavior of gulp plugins.

You can read some articles about error handling:
- [Error management in gulp](https://gist.github.com/floatdrop/8269868)
- [gulp-coffee's error handling document](https://github.com/wearefractal/gulp-coffee#error-handling)

So you have to handle errors by `on("error", ...)` or using [gulp-plumber](https://github.com/floatdrop/gulp-plumber) like below:

```
var gulp = require('gulp'),
    plumber = require('gulp-plumber'),
    typescript = require('gulp-tsc');

gulp.task('default', function () {
    gulp.watch('src/**/*.ts', ['compile'])
});

gulp.task('compile', function () {
    return gulp.src('src/**/*.ts')
        .pipe(plumber())
        .pipe(typescript())
        .pipe(gulp.dest('dest/'));
});
```

Also, some plugins like [gulp-sass](https://github.com/dlmanning/gulp-sass#errlogtoconsole-true) support option to not throwing errors on comilation failure.

I think gulp-tsc should support that option as well so that you don't have to install extra plugin or handle errors on yourself.

**Comment by kotas** [comment]
I have published v0.2.0 of gulp-tsc which has `emitError` option added.

You can pass `{ emitError: false }` to gulp-tsc to suppress error events and keep running gulp.watch task on compile errors.

```
var gulp = require('gulp'),
    typescript = require('gulp-tsc');

gulp.task('default', function () {
    gulp.watch('src/**/*.ts', ['compile'])
});

gulp.task('compile', function () {
    return gulp.src('src/**/*.ts')
        .pipe(typescript({ emitError: false }))
        .pipe(gulp.dest('dest/'));
});
```

**Comment by bazo** [comment]
thanks, works great
